### ISSUE TITLE
Using inbuilt pager like less/more for long outputs 

### ISSUE THREAD

**Issue by tsathis** [nushell/nushell#6046]
### Related problem

When the output of a shell command is longer than the height of the terminal window, what we have to do is scroll the terminal up until we find the starting line of output. Sometimes, the output might be even truncated when the output is very longer. 

What I normally do is running the command again with the `| less` or `| more`. However, the command may be costly or the output may be different in the second execution. Or, also I would use `| tee file.log` and read the file with a pager. 

Possible solution is to wrap the command in a function which pipes command output to less as mentioned in this [SO question](https://stackoverflow.com/questions/4174327/how-to-automatically-pipe-to-less-if-the-result-is-more-than-a-page-on-my-shell). However, it has two problems:

- We need to to write wrappers for each command we use
- Less doesn't have color outputs
    ![ss](https://user-images.githubusercontent.com/34299693/178895031-07ca62aa-7960-4dfa-bf15-d47cdcfeac2b.png)
 

Related: #389 #999 

### Describe the solution you'd like

- When I execute a command, the output need to be limited to the terminal windows height 
- Any further output can be viewed by scrolling the result (as like in `more` or `less`) 
- If the output is not longer than it must behave like normal way
- The color must be preserved
- If the input command is still visible on the top while scrolling the output it will be great
- It should work for all commands (not only for `ls` or table outputs)

### Additional context and details

 [How to automatically pipe to less if the result is more than a page on my shell?](https://stackoverflow.com/questions/4174327/how-to-automatically-pipe-to-less-if-the-result-is-more-than-a-page-on-my-shell)

**Comment by tharindusathis** [comment]
@fdncred thanks for the trick! Piping to `table` is important here. Is there any way to reuse the the output of the last command? If so we could simply create an alias for `$LastOutput | table | less -FRX`  and use. 
I think there's not such thing like `$LastOutput`, so I may just need to rerun the last command using `!! | table | less -FRX`. But, can we create alias with `!!`? Seems like `!!` is replaced in-place and cannot be used in a script.

**Comment by fdncred** [comment]
- You're welcome. :)
- There is no `$LastOutput` yet, although we've talked about it for some time.
- I don't believe `!!` will work in an alias or custom command, but I haven't tried either.
- You might be able to rig something up with `history` to get the last command and modify it?

**Comment by tharindusathis** [comment]
> * You might be able to rig something up with `history` to get the last command and modify it?

Yeah. That seems to be the way, thanks for the idea!

**Comment by wmstack** [comment]
Yes we need this. Windows does not have a less command, so it would be very helpful if an inbuilt pager was supported and used by default. Make it so that if the output of a command is larger than can fit in the screen, then open the pager by default.

**Comment by fdncred** [comment]
@wmstack If you download the nushell installer msi from the nushell repo and install it, you'll have the windows less.exe command. We bundle it with the Windows installer.

**Comment by jonnieey** [comment]
built nushell and installed nushell from git (using bash install-all.sh script ) changed shell to /usr/bin/nu (chsh -s /usr/bin/nu) $USER

Using arch linux,  st terminal (suckless)

> help commands | table | less -FRX
![2022-07-18-135334](https://user-images.githubusercontent.com/47189744/179497369-87e2548d-1f70-4f01-9669-83d743ba36c6.png)

**Comment by wmstack** [comment]
> @wmstack If you download the nushell installer msi from the nushell repo and install it, you'll have the windows less.exe command. We bundle it with the Windows installer.

Yeah, but the installer requires admin permissions to use, and so I am stuck with the scoop version right now. And I better have a pager!

**Comment by fdncred** [comment]
> Yeah, but the installer requires admin permissions ...

You can just download the msi and extract it without installing it. I do this all the time. I have a .reg file that I found on the internet where i right click on a msi file in explorer and then click extract. Works nice.

> Is there any reason that the less command is not built-in into the scoop installation with the other commands?

They must make their own msi. If they're using ours, it's in there. I'm not sure how scoop works. Scoop nushell installer is maintained but other people, not us.

**Comment by jonnieey** [comment]
> @jonnieey It looks like you have a font that doesn't support line drawing and your table_mode is set to rounded.

Thanks. My locale variables were causing the issue, `let-env LC_ALL=en_US.UTF-8` solved the issue

**Comment by wmstack** [comment]
Hello, I am getting the same garbled messages when using Windows Terminal with `less`:
```
# also tried with -RFX
help commands | table | less
```
![image](https://user-images.githubusercontent.com/58462210/181481259-640fc3e0-86a3-404c-a03e-be84045b53cc.png)

**Comment by fdncred** [comment]
@wmstack From my exploration, you have to have a specific version of `less`, actually, that version or later. I want to say 5.90 but I could be remembering wrong. Whatever we distribute is the "right" one though.

You also have to be using a font that support line drawing.

Apparently we learned above that env vars like LC_ALL can manipulate the less output too.

Do you know that you have met these conditions?

**Comment by wmstack** [comment]
* As for the font, the `ls` command by itself works naturally, so wouldn't that be excluded? I also tried it in both  Windows Terminal and in Vscode, and it is garbled in both.
* less version is 590 without dots:
```
less 590 (Spencer V8 regular expressions)
Copyright (C) 1984-2021  Mark Nudelman

less comes with NO WARRANTY, to the extent permitted by law.
For information about the terms of redistribution,
see the file named README in the less distribution.
Home page: https://greenwoodsoftware.com/less
```
* I don't appear to have an environment variable called LC_ALL on scanning through the autocompletion of `$env`

Another Windows user might need to verify this problem.

**Comment by fdncred** [comment]
Ya, I think if `ls` is working fine then your font is probably fine.

I'm running Windows today and I'm also using the exact same version of less.exe you're using. I'm not having issues.
![image](https://user-images.githubusercontent.com/343840/181780010-7f5fbb66-2f09-43d9-99e9-a5190362d954.png)

I'm wondering if your code page is messed up. You might try changing your code page to something like UTF8 which done with this command `chcp 65001`. I've used 437 too though. Here's a [link ](https://ss64.com/nt/chcp.html)to some other code pages.

I also have my env set like `let-env LESS = "-FRX"`
![image](https://user-images.githubusercontent.com/343840/181780883-6bef2bac-20ef-4936-bd76-65084fec8a4c.png)
and pager is set too and of course `less.exe` is in my path
![image](https://user-images.githubusercontent.com/343840/181780966-1cd6afc1-ab0d-479a-a528-edf2f3e00f0e.png)

I'm not sure what else it could be.

**Comment by wmstack** [comment]
Setting the code page to `65001` fixed it for me. Not sure why the `437` code is garbled. Could it be because we use different fonts?

**Comment by fdncred** [comment]
maybe code page 437 doesn't support line drawing?

**Comment by wmstack** [comment]
> maybe code page 437 doesn't support line drawing?

Is there a chance that Nushell gets a dedicated troubleshooting section in its documentation, for issues that have been faced before?

**Comment by wmstack** [comment]
Do you get those breaks in the lines?
![image](https://user-images.githubusercontent.com/58462210/181856580-cb4b97a9-90d7-4066-91ef-92d8c3333bec.png)

**Comment by unrelentingtech** [comment]
A built-in table pager could also do something `less` absolutely can't: **sticky headers**! :)

**Comment by fdncred** [comment]
@unrelentingtech @elferherrera and I have been looking and wishing for something like csvlens https://github.com/YS-L/csvlens

**Comment by remmycat** [comment]
> A built-in table pager could also do something `less` absolutely can't: **sticky headers**! :)

While I'm absolutely for having a built-in way to do this, this isn't true anymore!
Recent versions of less have a `--header` flag.
So `| table | less --header 3` is what you're looking for.

(On macOS I first had to install an up-to-date version of less via brew)

**Comment by fdncred** [comment]
We also have a pager PR this n process now that uses a tui.

**Comment by okbob** [comment]
I wrote support for nushell to pspg pager https://okbob.blogspot.com/2022/11/pspg-and-nushell.html

**Comment by AucaCoyan** [comment]
since `explore` was merged in #6984  I think this can be closed, right?

**Comment by fdncred** [comment]
less and explore work as well as amtoine's nu_plugin_explore
